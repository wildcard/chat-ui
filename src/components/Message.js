import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import classNames from 'classnames';
import Chip from 'material-ui/Chip';
import Typography from 'material-ui/Typography';
import ChatAvatar from './Avatar';
import { MessagePropType } from '../prop-types';

const styles = theme => ({
  root: {
    display: 'flex',
    justifyContent: 'center',
    flexWrap: 'wrap',
  },
  message: {
    margin: theme.spacing.unit,
  },
  selfMessage: {
    background: 'green'
  },
});

function Message(props) {
  const {
    classes,
    username,
    text,
    avatar,
    isSelf,
    ...other,
  } = props;
  return (<div className={classes.root} {...other}>
        <div>
          <ChatAvatar tiny src={avatar} username={username}/>
          <Typography gutterBottom noWrap>{username}</Typography>
        </div>
        <Chip label={text} className={classNames(classes.message, {
          [classes.selfMessage]: isSelf
        })} />
    </div>);
}

Message.propTypes = {
  // classes: PropTypes.object.isRequired,
  ...MessagePropType,
};

export default withStyles(styles)(Message);
