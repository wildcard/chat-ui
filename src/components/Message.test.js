import React from 'react';
import Message from './Message';
import renderer from 'react-test-renderer';

describe('snapshot', () => {
  it('', () => {
    const component = renderer.create(
      <Message username="kobi.kadosh" text="test message" />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('avatar', () => {
    const component = renderer.create(
      <Message
        avatar="https://s3.amazonaws.com/uifaces/faces/twitter/jeremiespoken/128.jpg"
        username="kobi.kadosh"
        text="test message"
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('isMe', () => {
    const component = renderer.create(
      <Message isMe username="kobi.kadosh" text="test message" />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('isMe is differnt than regular message design', () => {
    const myMessage = renderer.create(
      <Message isMe username="kobi.kadosh" text="test message" />
    );
    const message = renderer.create(
      <Message isMe username="kobi.kadosh" text="test message" />
    );

    let treeA = myMessage.toJSON();
    let treeB = message.toJSON();

    expect(treeA).not.toEqual(treeB);
  });

  it('isMe + prevMessageIsSameUser', () => {
    const component = renderer.create(
      <Message
        isMe
        prevMessageIsSameUser
        username="kobi.kadosh"
        text="test message"
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('prevMessageIsSameUser', () => {
    const component = renderer.create(
      <Message
        prevMessageIsSameUser
        username="kobi.kadosh"
        text="test message"
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('isLast', () => {
    const component = renderer.create(
      <Message isLast username="kobi.kadosh" text="test message" />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
