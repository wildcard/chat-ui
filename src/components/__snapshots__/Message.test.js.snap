// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot  1`] = `
<div
  className="Message-root-1"
>
  <div
    className="Message-messageContainer-3"
  >
    <div
      className="Message-messageHead-4"
    >
      <div
        className="MuiAvatar-root-12 ChatAvatar-tiny-10 ChatAvatar-avatar-9"
      >
        <img
          alt="kobi.kadosh"
          className="MuiAvatar-img-14"
          sizes={undefined}
          src="https://spotim-demo-chat-server.herokuapp.com/avatars/003-pikachu.png"
          srcSet={undefined}
        />
      </div>
      <p
        className="MuiTypography-root-15 MuiTypography-body1-24 MuiTypography-noWrap-31 MuiTypography-gutterBottom-32"
      >
        kobi.kadosh
      </p>
    </div>
    <div
      className="MuiChip-root-39 Message-messageTextRoot-7 Message-message-5"
      onClick={undefined}
      onKeyDown={[Function]}
      role="button"
      tabIndex={-1}
    >
      <span
        className="MuiChip-label-44 Message-messageTextLabel-8"
      >
        test message
      </span>
    </div>
  </div>
</div>
`;

exports[`snapshot avatar 1`] = `
<div
  className="Message-root-1"
>
  <div
    className="Message-messageContainer-3"
  >
    <div
      className="Message-messageHead-4"
    >
      <div
        className="MuiAvatar-root-12 ChatAvatar-tiny-10 ChatAvatar-avatar-9"
      >
        <img
          alt="kobi.kadosh"
          className="MuiAvatar-img-14"
          sizes={undefined}
          src="https://s3.amazonaws.com/uifaces/faces/twitter/jeremiespoken/128.jpg"
          srcSet={undefined}
        />
      </div>
      <p
        className="MuiTypography-root-15 MuiTypography-body1-24 MuiTypography-noWrap-31 MuiTypography-gutterBottom-32"
      >
        kobi.kadosh
      </p>
    </div>
    <div
      className="MuiChip-root-39 Message-messageTextRoot-7 Message-message-5"
      onClick={undefined}
      onKeyDown={[Function]}
      role="button"
      tabIndex={-1}
    >
      <span
        className="MuiChip-label-44 Message-messageTextLabel-8"
      >
        test message
      </span>
    </div>
  </div>
</div>
`;

exports[`snapshot isLast 1`] = `
<div
  className="Message-root-1"
>
  <div
    className="Message-messageContainer-3"
  >
    <div
      className="Message-messageHead-4"
    >
      <div
        className="MuiAvatar-root-12 ChatAvatar-tiny-10 ChatAvatar-avatar-9"
      >
        <img
          alt="kobi.kadosh"
          className="MuiAvatar-img-14"
          sizes={undefined}
          src="https://spotim-demo-chat-server.herokuapp.com/avatars/003-pikachu.png"
          srcSet={undefined}
        />
      </div>
      <p
        className="MuiTypography-root-15 MuiTypography-body1-24 MuiTypography-noWrap-31 MuiTypography-gutterBottom-32"
      >
        kobi.kadosh
      </p>
    </div>
    <div
      className="MuiChip-root-39 Message-messageTextRoot-7 Message-message-5"
      onClick={undefined}
      onKeyDown={[Function]}
      role="button"
      tabIndex={-1}
    >
      <span
        className="MuiChip-label-44 Message-messageTextLabel-8"
      >
        test message
      </span>
    </div>
  </div>
</div>
`;

exports[`snapshot isMe + prevMessageIsSameUser 1`] = `
<div
  className="Message-root-1 Message-rootSelf-2"
>
  <div
    className="Message-messageContainer-3"
  >
    <div
      className="MuiChip-root-39 Message-messageTextRoot-7 Message-message-5 Message-selfMessage-6"
      onClick={undefined}
      onKeyDown={[Function]}
      role="button"
      tabIndex={-1}
    >
      <span
        className="MuiChip-label-44 Message-messageTextLabel-8"
      >
        test message
      </span>
    </div>
  </div>
</div>
`;

exports[`snapshot isMe 1`] = `
<div
  className="Message-root-1 Message-rootSelf-2"
>
  <div
    className="Message-messageContainer-3"
  >
    <div
      className="MuiChip-root-39 Message-messageTextRoot-7 Message-message-5 Message-selfMessage-6"
      onClick={undefined}
      onKeyDown={[Function]}
      role="button"
      tabIndex={-1}
    >
      <span
        className="MuiChip-label-44 Message-messageTextLabel-8"
      >
        test message
      </span>
    </div>
  </div>
</div>
`;

exports[`snapshot prevMessageIsSameUser 1`] = `
<div
  className="Message-root-1"
>
  <div
    className="Message-messageContainer-3"
  >
    <div
      className="MuiChip-root-39 Message-messageTextRoot-7 Message-message-5"
      onClick={undefined}
      onKeyDown={[Function]}
      role="button"
      tabIndex={-1}
    >
      <span
        className="MuiChip-label-44 Message-messageTextLabel-8"
      >
        test message
      </span>
    </div>
  </div>
</div>
`;
